docs/cookbook/squad_extraction.md

```diff
@@ -0,0 +1,187 @@
+# SQuAD 2.0: Extracting Answers To Questions From Context Using Mirascope
+
+The SQuAD 2.0 Dataset is a dataset for question-answering. Each question is either impossible to answer given the context paragraph or has answers exactly as written in the paragraph. It turns out that it's quite difficult to get an LLM to ignore it's world knowledge and output "<No Answer>" if the paragraph does not contain the answer. For this recipe, we'll be using a modified version of the dataset, restricting our questions to a single article (Geology) and removing all of the impossible questions.
+
+!!! note
+
+    The [`geology-squad.json`](https://raw.githubusercontent.com/Mirascope/datasets/main/geology-squad.json) modified version of the dataset is on our GitHub along with the [full code](https://github.com/Mirascope/mirascope/tree/main/cookbook/extraction_examples/squad_extraction) for this recipe.
+
+The first step is loading the dataset. We've written a helper function `load_geology_squad` that will load and return a `list[QuestionWithContext]`, defined below:
+
+```python
+from pydantic import BaseModel
+
+class QuestionWithContext(BaseModel):
+    """A question with it's answers and context."""
+
+    id: str
+    question: str
+    context: str
+    answers: list[str]
+```
+
+## Initial Basic Extraction
+
+To extract an answer to a question, we can use the [`OpenAIChat.extract`](../api/chat/models/openai_chat.md/#mirascope.chat.models.OpenAIChat.extract) method to extract the answer. First, let's create a super basic schema and prompt for asking the question and extracting an answer:
+
+```shell
+$ mirascope init; touch prompts/question.py
+```
+
+```python
+# prompts/question.py
```

Comment:
in our cookbook example, you used squad_prompts, should we match them?

Reply:
Good catch.